src/classes/USER_UserService.cls

```diff
@@ -0,0 +1,271 @@
+    * @description Determines if each User in the List is active based on IsActive field.
+    * @param updatedUsers The users that were modified.
+    * @return A Map containing the UserId and Boolean value based on the User's IsActive status.
+    **********************************************************************************************************/
+	public static Map<Id, Boolean> isActiveUser(List<User> updatedUsers) {
+		Map<Id, Boolean> userIdToIsActive = new Map<Id, Boolean>();
+		
+		// Determine if the User isActive
+		if(null != updatedUsers && !updatedUsers.isEmpty()) {
+			for(User currentUser : updatedUsers) {
+				if(currentUser.IsActive) {
+					userIdToIsActive.put(currentUser.Id, true);
+				} else {
+					userIdToIsActive.put(currentUser.Id, false);
+				}
+			}
+		}
+
+		return userIdToIsActive;
+	}
+
+
+	/*********************************************************************************************************
+    * @description Determines if each User in the List is frozen based on IsFrozen field.
+    * @param updatedUsers The user(s) that were modified.
+    * @return A Map containing the UserId and Boolean value based on the User's IsFrozen status.
+    **********************************************************************************************************/
+	public static Map<Id, Boolean> isFrozenUser(List<User> updatedUsers) {
+		Map<Id, Boolean> userIdToIsFrozen = new Map<Id, Boolean>();
+
+		// Retrieve the UserLogin records for the updatedUsers
+		if(null != updatedUsers && !updatedUsers.isEmpty()) {
+			Map<Id, Boolean> userLoginUserIdToIsFrozen = retrieveUserLogins(updatedUsers);
+			
+			// Populate the Map with UserId and IsFrozen
+			for(User currentUser : updatedUsers) {
+				userIdToIsFrozen.put(currentUser.Id, userLoginUserIdToIsFrozen.get(currentUser.Id));
+			}
+		}
+
+		return userIdToIsFrozen;
+	}
+	
+
+	/*********************************************************************************************************
+    * @description Handles User changes to IsActive/IsFrozen status by updating the Error Notification Recipients, logging an Error__c, and notifying the org's System Administrators.
+    * @param sysAdminUserIds The User Ids that were modified.
+    * @return void
+    **********************************************************************************************************/
+	@future
+	public static void applySysAdminIsActiveIsFrozenStatusChanges(Set<Id> sysAdminUserIds) {
+		// Sets Error Notification recipient to All Sys Admin
+		setErrorNotificationToAllSysAdmins(sysAdminUserIds);
+
+		// Logs an Error and Emails the Sys Admins
+		handleScheduledJobsSubmittedByInActiveIsFrozenSysAdmin(sysAdminUserIds);
+	}
+
+
+	/*********************************************************************************************************
+    * @description Sets the NPSP Settings Error Notification Recipients to All Sys Admins.
```

Comment:
does it always set the recipients to All Sys Admins, or only if one of the users made inactive/frozen is the current recipient?  I believe the latter, and your description should make this clear.  I'd prefer a method name that also didn't imply the action was always taken.  While reviewing the code in the callers, I thought it was wrong because it didn't check if the current user was the only recipient.  maybe a name like handleErrorRecipientInactiveIsFrozen() or something like that?

Reply:
Updated the method's name.